### PR TITLE
Assessment test

### PR DESCRIPTION
**Pull request by tijjvni**
Task reference

This section contains a link to a corresponding task in Linear.
Summary

Optimized and tested the assessment flow
This section contains a brief description of what was accomplished within the current PR.
Implementation details

Wrote unit tests to test all the assessment endpoints
This section describes the technical aspects of the solution, implemented withing the current PR, that are worth mentioning to the reviewers.
-Created Factory for consumption and fake creating of models for use in the unit testst
How to test
url: [https://api.eval360.hng.tech/api/assessment](https://api.eval360.hng.tech/api/assessment](https://api.eval360.hng.tech/api/assessment%5D(https://api.eval360.hng.tech/api/assessment)

This section contains a checklist, describing basic scenarios that would help verify that the submitted code works correctly.
References

Any additional information (screenshots, video, diagrams) that would help understand and review the PR.

https://user-images.githubusercontent.com/81570975/206475317-0af66e8f-38cb-4770-8045-37499ed765d0.png
Screenshot 2022-12-08 153357

### CODE REVIEW

File: backend/app/Http/Controllers/AssessmentController.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function createAssessment(Request $request)
 
             Assessment::create($data);
 
-            return $this->sendResponse(false, null, "assessment created.", $data, 200);
+            return $this->sendResponse(false, null, "assessment created.", $data, Response::HTTP_CREATED);
         } catch (\Exception $e) {
             return $this->sendResponse(true, 'something went wrong creating assessment', $e->getMessage(), null, 500);
```
Comment (dasaint001) [inline backend/app/Http/Controllers/AssessmentController.php]: http response still missing here

File: backend/app/Http/Controllers/AssessmentController.php [hunk 2]
```diff
@@ -106,7 +106,7 @@ public function updateAssessment(Request $request, $assessmentId)
 
             $assessment->update($updatedData);
 
-            return $this->sendResponse(false, null, 'assessment updated successfully', $updatedName, Response::HTTP_OK);
+            return $this->sendResponse(false, null, 'assessment updated successfully', $assessment->first()->refresh(), Response::HTTP_OK);
         } catch (Exception $e) {
             return $this->sendResponse(true, "something went wrong updating assessment " . $e->getMessage(), 'failed updating assessment.', null, 500);
```
Comment (dasaint001) [inline backend/app/Http/Controllers/AssessmentController.php]: same here

**Review by dasaint001 — CHANGES_REQUESTED** [review]

File: backend/app/Http/Controllers/AssessmentController.php (hunk 1 above)
Comment (tijjvni) [reply backend/app/Http/Controllers/AssessmentController.php]: Which http response are you talking about? I think it is passed on line 53

Comment (dasaint001) [reply backend/app/Http/Controllers/AssessmentController.php]: line 55

File: backend/app/Http/Controllers/AssessmentController.php (hunk 2 above)
Comment (dasaint001) [reply backend/app/Http/Controllers/AssessmentController.php]: line 111

File: backend/app/Http/Controllers/AssessmentController.php (hunk 1 above)
Comment (tijjvni) [reply backend/app/Http/Controllers/AssessmentController.php]: Fixed

**Comment by dasaint001** [comment]
not needed

**Pull request closed without merging**